gadget/validate.go

```diff
@@ -239,42 +278,12 @@ func validateVolumeContentsPresence(gadgetSnapRootDir string, vol *LaidOutVolume
 	return nil
 }
 
-func validateEncryptionSupport(info *Info) error {
-	for name, vol := range info.Volumes {
-		var haveSave bool
-		for _, s := range vol.Structure {
-			if s.Role == SystemSave {
-				haveSave = true
-			}
-		}
-		if !haveSave {
-			return fmt.Errorf("volume %q has no structure with system-save role", name)
-		}
-		// XXX: shall we make sure that size of ubuntu-save is reasonable?
-	}
-	return nil
-}
-
-type ValidationConstraints struct {
-	// EncryptedData when true indicates that the gadget will be used on a
-	// device where the data partition will be encrypted.
-	EncryptedData bool
-}
-
-// Validate checks whether the given directory contains valid gadget snap
-// metadata and a matching content, under the provided model constraints, which
-// are handled identically to ReadInfo(). Optionally takes additional validation
-// constraints, which for instance may only be known at run time,
-func Validate(gadgetSnapRootDir string, model Model, extra *ValidationConstraints) error {
+// ValidateContent checks whether the given directory contains valid matching content with respect to the given pre-validated gadget metadata.
+func ValidateContent(info *Info, gadgetSnapRootDir string) error {
 	// TODO: also validate that only one "<bl-name>.conf" file is in the root
 	//       directory  of the gadget snap, because the "<bl-name>.conf" file
```

Comment:
my typo really, but this has extra whitespace
```suggestion
	//       directory of the gadget snap, because the "<bl-name>.conf" file
```